plugins/com.contrastsecurity.ide.eclipse.ui/src/com/contrastsecurity/ide/eclipse/ui/internal/views/FilterDialog.java

```diff
@@ -0,0 +1,714 @@
+				dateTimeFrom.setMonth(localDateTimeMinusWeek.getMonthValue() - 1);
+				dateTimeFrom.setHours(localDateTimeMinusWeek.getHour());
+				dateTimeFrom.setMinutes(localDateTimeMinusWeek.getMinute());
+				dateTimeFrom.setSeconds(localDateTimeMinusWeek.getSecond());
+				break;
+			case Constants.LAST_DETECTED_MONTH:
+				LocalDateTime localDateTimeMinusMonth = localDateTime.minusMonths(1);
+
+				dateTimeFrom.setYear(localDateTimeMinusMonth.getYear());
+				dateTimeFrom.setDay(localDateTimeMinusMonth.getDayOfMonth());
+				dateTimeFrom.setMonth(localDateTimeMinusMonth.getMonthValue() - 1);
+				dateTimeFrom.setHours(localDateTimeMinusMonth.getHour());
+				dateTimeFrom.setMinutes(localDateTimeMinusMonth.getMinute());
+				dateTimeFrom.setSeconds(localDateTimeMinusMonth.getSecond());
+				break;
+			case Constants.LAST_DETECTED_YEAR:
+				LocalDateTime localDateTimeMinusYear = localDateTime.minusYears(1);
+
+				dateTimeFrom.setYear(localDateTimeMinusYear.getYear());
+				dateTimeFrom.setDay(localDateTimeMinusYear.getDayOfMonth());
+				dateTimeFrom.setMonth(localDateTimeMinusYear.getMonthValue() - 1);
+				dateTimeFrom.setHours(localDateTimeMinusYear.getHour());
+				dateTimeFrom.setMinutes(localDateTimeMinusYear.getMinute());
+				dateTimeFrom.setSeconds(localDateTimeMinusYear.getSecond());
+				break;
+			case Constants.LAST_DETECTED_CUSTOM:
+				dateTimeFrom.setEnabled(true);
+				dateTimeTo.setEnabled(true);
+				break;
+			}
+		}
+	};
+
+	public FilterDialog(Shell parentShell, ContrastSDK contrastSDK, Servers servers, Applications applications) {
+		super(parentShell);
+
+		extendedContrastSDK = (ExtendedContrastSDK) contrastSDK;
+		this.servers = servers;
+		this.applications = applications;
+	}
+
+	private String getOrgUuid() {
+		String orgUuid = null;
+		try {
+			orgUuid = ContrastCoreActivator.getSelectedOrganizationUuid();
+		} catch (Exception e) {
+			ContrastUIActivator.log(e);
+		}
+		return orgUuid;
+	}
+
+	private void createApplicationCombo(Composite composite, String orgUuid) {
+		applicationCombo = new ComboViewer(composite, SWT.READ_ONLY);
+		applicationCombo.getControl().setFont(composite.getFont());
+		applicationCombo.setLabelProvider(new ContrastLabelProvider());
+		applicationCombo.setContentProvider(new ArrayContentProvider());
+	}
+
+	private void createServerCombo(Composite composite, String orgUuid) {
+		serverCombo = new ComboViewer(composite, SWT.READ_ONLY);
+		serverCombo.getControl().setFont(composite.getFont());
+		serverCombo.setLabelProvider(new ContrastLabelProvider());
+		serverCombo.setContentProvider(new ArrayContentProvider());
+	}
+
+	private void createLastDetectedCombo(Composite composite) {
+		lastDetectedCombo = new ComboViewer(composite, SWT.READ_ONLY);
+		lastDetectedCombo.getControl().setFont(composite.getFont());
+		lastDetectedCombo.setLabelProvider(new LabelProvider());
+		lastDetectedCombo.setContentProvider(new ArrayContentProvider());
```

Comment:
Create a function for the combo initialization in order to avoid repeating code, since they all receive the same parameters

```
private void createServerCombo(Composite composite, String orgUuid) {
     serverCombo = initComboViewer(composite);
}

private ComboViewer initComboViewer(Composite composite) {
     ComboViewer = new ComboViewer(composite, SWT.READ_ONLY);
     //More code...
     return viewer;
}
```

Reply:
Done.